client/pluginManger.go

```diff
@@ -133,6 +139,53 @@ func (p *PluginManager) InstallPlugin(names []string) (err error) {
 	return
 }
 
+func (p *PluginManager) writeFileLocally(statusCode int, body []byte, name string) (err error) {
+	if statusCode != 200 {
+		err = fmt.Errorf("unexpected status code: %d", statusCode)
+		if p.Debug {
+			ioutil.WriteFile(debugLogFile, body, 0664)
+		}
+		return err
+	}
+	
+	if err = ioutil.WriteFile(fmt.Sprintf("%s.hpi", name), body, 0644); err != nil {
+		if p.Debug {
+			ioutil.WriteFile(debugLogFile, []byte(err.Error()), 0664)
```

Comment:
Consider putting these error handles into a method. So we don't need to offer test cases for everyone.